package/MDAnalysis/analysis/hole.py

```diff
@@ -356,6 +362,7 @@ def plot(self, **kwargs):
         ax.set_ylabel(r"HOLE radius $r$ ($\AA$)")
         if kw['label']:
             ax.legend(loc="best")
+        return ax
```

Comment:
Actually, there is one: if there hadn't been an axis before, `plt.gca()` will create one. In what I propose, it would be created even though we don't need it. It's also not closed again so it would use up resources that we don't need. I will stick with your original code.